packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -142,43 +143,54 @@ export const useCheckoutPage = props => {
     }, [checkoutStep, setCheckoutStep]);
 
     const handlePlaceOrder = useCallback(async () => {
-        try {
-            await getOrderDetails({
-                variables: {
-                    cartId
-                }
-            });
-            await placeOrder({
-                variables: {
-                    cartId
-                }
-            });
-
-            await removeCart();
-
-            await clearCartDataFromCache(apolloClient);
-
-            await createCart({
-                fetchCartId
-            });
-        } catch (err) {
-            console.error(
-                'An error occurred during when placing the order',
-                err
-            );
-            setReviewOrderButtonClicked(false);
-            setCheckoutStep(CHECKOUT_STEP.PAYMENT);
-            // TODO: Delete nonce? The nonce might be expired and why the order
-            // failed. If we delete it the payment info section will render as
-            // if it was not filled, thus prompting the user to enter new info.
+        // Fetch order details and then use an effect to actually place the
+        // order. If/when Apollo returns promises for invokers from useLazyQuery
+        // we can just await this function and then perform the rest of order
+        // placement.
```

Comment:
Yeah, the [Apollo docs](https://www.apollographql.com/docs/react/api/react-hooks/#result-1) say the following about the invoker:

> Function that can be triggered to execute the suspended query. After being called, `useLazyQuery` behaves just like `useQuery`.

So I think they do intend for us to use `called`, `data`, and `loading` as returned by the hook, since they're updated by the hook itself as part of its internal lifecycle. And if we want to use the hook's return values, we have to do it as an effect, just as you've done here.